homeassistant/components/atome/sensor.py

```diff
@@ -125,9 +128,53 @@ def update_live_usage(self):
                 self._is_connected,
                 self._subscribed_power,
             )
+        else:
+            _LOGGER.error("Live Data : Missing last value in values: %s", values)
+            error_during_retrieve = True
+        return not error_during_retrieve
 
-        except KeyError as error:
-            _LOGGER.error("Missing last value in values: %s: %s", values, error)
+    @Throttle(LIVE_SCAN_INTERVAL)
+    def update_live_usage(self):
+        """Return current power value."""
+        retrieve_success = self._retrieve_live()
+
+        if not retrieve_success:
+            _LOGGER.debug("Perform Reconnect during live request")
+            self.atome_client.login()
+            self._retrieve_live()
+
+    def _retrieve_period_usage(self, period_type):
+        """Return current daily/weekly/monthly/yearly power usage."""
+        error_during_retrieve = False
+        period_usage = None
+        period_price = None
+        values = self.atome_client.get_consumption(period_type)
+        if values.get("total") and values.get("price"):
+            period_usage = values["total"] / 1000
+            period_price = values["price"]
+            _LOGGER.debug("Updating Atome %s data. Got: %d", period_type, period_usage)
+        else:
+            error_during_retrieve = True
+            _LOGGER.error("%s : Missing last value in values: %s", period_type, values)
+        retrieve_success = not error_during_retrieve
+        return retrieve_success, period_usage, period_price
```

Comment:
```suggestion
    def _retrieve_period_usage(self, period_type):
        """Return current daily/weekly/monthly/yearly power usage."""
        values = self.atome_client.get_consumption(period_type)
        if values.get("total") and values.get("price"):
            period_usage = values["total"] / 1000
            period_price = values["price"]
            _LOGGER.debug("Updating Atome %s data. Got: %d", period_type, period_usage)
            return True, period_usage, period_price
        _LOGGER.error("%s : Missing last value in values: %s", period_type, values)
        return False, None, None
```

Reply:
ok , I'll make it shorter.
(I will add comments for maintanability)